src/utils.ts

```diff
@@ -0,0 +1,410 @@
+    if (!network) {
+      throw new Error('network not found');
+    }
+    address.toOutputScript(addr, network);
+    return true;
+  } catch (ignore) {
+    return false;
+  }
+};
+
+export async function signMessageWithMnemonic(
+  message: string,
+  mnemonic: string,
+  network: networks.Network
+): Promise<SignedMessage> {
+  const seed = await mnemonicToSeed(mnemonic);
+  const node = bip32.fromSeed(seed, network);
+  const child = node.derivePath("m/84'/0'/0'/0/0");
+  const signature = await signAsync(message, child.privateKey!, true, {
+    segwitType: 'p2wpkh',
+  });
+
+  const pay = payments.p2wpkh({ pubkey: child.publicKey, network });
+  return {
+    signature: signature.toString('base64'),
+    address: pay.address!,
+    publicKey: child.publicKey.toString('hex'),
+  };
+}
+
+export function computeBalances(utxos: UnblindedOutput[]): Record<string, number> {
+  const balances: Record<string, number> = {};
+  for (const utxo of utxos) {
+    if (!utxo.blindingData) continue;
+    const { asset, value } = utxo.blindingData;
+    balances[asset] = (balances[asset] || 0) + value;
+  }
+  return balances;
+}
+
+export function getNetwork(network: NetworkString): networks.Network {
+  const net = networks[network];
+  if (!net) {
+    throw new Error('network not found');
+  }
+  return net;
+}
+
+const reverseHex = (hex: string) => Buffer.from(hex, 'hex').reverse().toString('hex');
+
+export async function makeURLwithBlinders(transaction: Transaction) {
+  const webExplorerURL = await appRepository.getWebExplorerURL();
+  if (!webExplorerURL) {
+    throw new Error('web explorer url not found');
+  }
+  const txID = transaction.getId();
+
+  const blinders: string[] = [];
+  for (let i = 0; i < transaction.outs.length; i++) {
+    const output = transaction.outs[i];
+    if (output.script.length === 0) continue;
+    const data = await walletRepository.getOutputBlindingData(txID, i);
+    if (!data || !data.blindingData) continue;
+
+    blinders.push(
+      `${data.blindingData.value},${data.blindingData.asset},${reverseHex(
+        data.blindingData.valueBlindingFactor
+      )},${reverseHex(data.blindingData.assetBlindingFactor)}`
+    );
+  }
+
+  const url = `${webExplorerURL}/tx/${txID}#blinded=${blinders.join(',')}`;
+  return url;
+}
+
+function estimateScriptSigSize(type: ScriptType): number {
+  switch (type) {
+    case ScriptType.P2Pkh:
+      return 108;
+    case (ScriptType.P2Sh, ScriptType.P2Wsh):
+      return 35;
+    case (ScriptType.P2Wsh, ScriptType.P2Tr, ScriptType.P2Wpkh):
+      return 1;
+  }
+  return 0;
+}
+
+const INPUT_BASE_SIZE = 40; // 32 bytes for outpoint, 4 bytes for sequence, 4 for index
+
+function txBaseSize(inScriptSigsSize: number[], outNonWitnessesSize: number[]): number {
+  const inSize = inScriptSigsSize.reduce((a, b) => a + b + INPUT_BASE_SIZE, 0);
+  const outSize = outNonWitnessesSize.reduce((a, b) => a + b, 0) + 33 + 9 + 1 + 1; // add unconf fee output size
```

Comment:
Can use the constant `outNonWitnessesSize.reduce((a, b) => a + b, 0) + FEE_OUTPUT_SIZE`

Reply:
This should not be there, there is a new `withFeeOutput` argument handling that case in `estimateVirtualSize` function. fixed :heavy_check_mark: